app/reducers/surveys.ts

```diff
@@ -18,126 +20,113 @@ import type {
-        ({
-          ...surveysById[surveyId],
-          event: events.find(
-            (event) => event.id === surveysById[surveyId].event,
-          ),
-        }) as unknown as SelectedSurvey,
-    );
-  },
-);
+export default surveysSlice.reducer;
+
+export const {
+  selectAll: selectAllSurveys,
+  selectById: selectSurveyById,
+  selectByField: selectSurveysByField,
+} = legoAdapter.getSelectors((state: RootState) => state.surveys);
 
-export const selectSurveyById = createSelector(
-  (state: RootState) => selectSurveys(state),
-  (_: RootState, surveyId?: EntityId) => surveyId,
-  (surveys, surveyId) => {
-    return surveys.find((survey) => survey.id === Number(surveyId));
-  },
-);
 export const selectSurveyTemplates = createSelector(
-  (state: RootState) => selectSurveys(state),
+  selectAllSurveys,
   (surveys) => surveys.filter((survey) => survey.templateType),
 );
 
-type SurveyTemplateProps = {
-  templateType: EventType;
-};
-export type SurveyTemplate = Omit<
-  DetailedSurvey,
-  'id' | 'event' | 'activeForm' | 'templateType'
-> & {
-  questions: (Omit<DetailedSurvey['questions'][number], 'id'> & {
-    options: Omit<DetailedSurvey['questions'][number]['options'], 'id'>;
-  })[];
-  templateType: EventType;
-};
-
-export const selectSurveyTemplate = createSelector(
-  (state: RootState) => selectSurveys(state),
-  (_: RootState, props: SurveyTemplateProps) => props.templateType,
-  (surveys, templateType) => {
-    const template = surveys.find(
-      (survey) => survey.templateType === templateType,
-    );
-    if (!template) return undefined;
+export type TransformedSurveyTemplate = Overwrite<
+  Omit<DetailedSurvey, 'id' | 'event' | 'activeFrom'>,
+  {
+    questions: Overwrite<
+      Omit<SurveyQuestion, 'id'>,
+      {
+        options: Omit<SurveyQuestionOption, 'id'>[];
+      }
+    >[];
+  }
+>;
+export const selectSurveyTemplateByType = createSelector(
+  selectSurveysByField('templateType').single,
+  (template) => {
+    if (!template || !('actionGrant' in template)) return undefined;
     const questions = (template.questions || []).map((question) => ({
       ...omit(question, 'id'),
       options: question.options.map((option) => omit(option, 'id')),
     }));
     return {
       ...omit(template, ['id', 'event', 'activeFrom']),
       questions,
-    } as SurveyTemplate;
+    } satisfies TransformedSurveyTemplate;
   },
 );
 
 export const useFetchedTemplate = (
   prepareId: string,
   templateType?: EventType,
-): SurveyTemplate | undefined => {
+): TransformedSurveyTemplate | undefined => {
   const dispatch = useAppDispatch();
   usePreparedEffect(
     `useFetchedTemplate-${prepareId}`,
     () => templateType && dispatch(fetchTemplate(templateType)),
     [templateType],
   );
   return useAppSelector((state: RootState) =>
-    templateType ? selectSurveyTemplate(state, { templateType }) : undefined,
+    templateType ? selectSurveyTemplateByType(state, templateType) : undefined,
```

Comment:
This seems to be handled by `selectSurveyTemplateByType`

Reply:
Ah, yes:)